app/forms/mappings/Constraints.scala

```diff
@@ -71,6 +71,16 @@ trait Constraints extends EmailValidation {
         Invalid(errorKey, regex)
     }
 
+  protected def regexpDynamic(regex: String, errorKey: String, dynamicMessage: String)(implicit
```

Comment:
Can you add some simple unit tests for this in `ConstraintsSpec`

Reply:
Yep will do